### PR TITLE
21248 Fixed continuation authorization reactivity issue

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#21248

*Description of changes:*
- app = 5.10.16
- initialized rest of Continuation Authorization object variable so they are reactive

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/ContinuationIn/ContinuationAuthorization.vue [hunk 1]
```diff
@@ -191,8 +191,8 @@ export default class ExtraproRegistration extends Mixins(DocumentMixin) {
 
   /** Called when this component is mounted. */
   mounted (): void {
-    // set or initialize authorization object
-    this.authorization = this.getContinuationAuthorization || { files: [] } as ContinuationAuthorizationIF
+    this.authorization = this.getContinuationAuthorization ||
+      { files: [], date: null, expiryDate: null } as ContinuationAuthorizationIF
```
Comment (JazzarKarim) [inline src/components/ContinuationIn/ContinuationAuthorization.vue:195]: So, initially setting them to null fixed the reactivity issue?

**Review by JazzarKarim — APPROVED** [review]

File: src/components/ContinuationIn/ContinuationAuthorization.vue (hunk 1 above)
Comment (severinbeauvais) [inline src/components/ContinuationIn/ContinuationAuthorization.vue:195]: The reason for the Date and Expiry Date not being synced to the store and thus displayed on the review page is that the watcher on line 270 below didn't fire.

The reason that the watcher didn't fire, even though the "deep watch" option was enabled, is that the Date and Expiry Date were not reactive.

The reason that the Date and Expiry Date were not reactive is that they were undefined in the initial object (only `files` was initialized). Now that I initialize them, they are reactive and the watcher fires and the store is updated and the review step shows their value.

What line 194 does is set the local variable to point (refer) to the same object that is in the store. This is not a new memory allocation; it's the same object. So changing object properties in this component is actually setting those properties in the store object. However, if the store object was initially null then line 195 assigns a new object (memory) to the local variable. This is then synced to the store on line 272.

Comment (severinbeauvais) [reply src/components/ContinuationIn/ContinuationAuthorization.vue:195]: You got it.

Comment (JazzarKarim) [reply src/components/ContinuationIn/ContinuationAuthorization.vue:195]: I bet that took some time to identify. Sweet, thanks for the clarification Sev! It makes sense.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-698-h4ya0yrz.web.app

File: src/components/ContinuationIn/ContinuationAuthorization.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/ContinuationIn/ContinuationAuthorization.vue:195]: Remember, undefined local variables, and undefined object properties, are not reactive.

That's why, elsewhere, when I added properties to an object, I used `$set(object, property)`.

Comment (JazzarKarim) [reply src/components/ContinuationIn/ContinuationAuthorization.vue:195]: Keeping this in mind.

Comment (severinbeauvais) [reply src/components/ContinuationIn/ContinuationAuthorization.vue:195]: Nope, I figured out this bug right away.

But then, a while back, I spent time learning (the hard way) about undefined variables and properties 😆

**Pull request merged**